File: Signiture-Backend/delivery/db/src/main/java/pcy/study/db/userordermenu/UserOrderMenu.java
package pcy.study.db.userordermenu;

import jakarta.persistence.Column;
import jakarta.persistence.Entity;
import jakarta.persistence.EnumType;
import jakarta.persistence.Enumerated;
import lombok.AccessLevel;
import lombok.Builder;
import lombok.Getter;
import lombok.NoArgsConstructor;
import pcy.study.db.BaseTimeEntity;
import pcy.study.db.userordermenu.enums.UserOrderMenuStatus;

@Getter
@NoArgsConstructor(access = AccessLevel.PROTECTED)
@Entity
public class UserOrderMenu extends BaseTimeEntity {

    @Column(nullable = false)
    private Long userOrderId;

    @Column(nullable = false)
    private Long storeMenuId;

    @Column(length = 50, nullable = false, columnDefinition = "VARCHAR")
    @Enumerated(EnumType.STRING)
    private UserOrderMenuStatus status;

    @Builder
    private UserOrderMenu(Long userOrderId, Long storeMenuId) {
        this.userOrderId = userOrderId;
        this.storeMenuId = storeMenuId;
        this.status = UserOrderMenuStatus.REGISTERED;
    }
}

File: Signiture-Backend/kotlin/delivery/db/src/main/kotlin/pcy/study/db/userordermenu/UserOrderMenu.kt
package pcy.study.db.userordermenu

import jakarta.persistence.*
import org.hibernate.annotations.CreationTimestamp
import org.hibernate.annotations.UpdateTimestamp
import pcy.study.db.storemenu.StoreMenu
import pcy.study.db.userorder.UserOrder
import pcy.study.db.userordermenu.enums.UserOrderMenuStatus
import java.time.LocalDateTime

class UserOrderMenu @JvmOverloads constructor(

    @field:Id
    @field:GeneratedValue(strategy = GenerationType.AUTO)
    var id: Long? = null,

    @field:ManyToOne
    @field:JoinColumn(name = "user_order_id", nullable = false)
    var userOrder: UserOrder,

    @field:ManyToOne
    @field:JoinColumn(name = "store_menu_id", nullable = false)
    var storeMenu: StoreMenu,

    @field:Column(length = 50, nullable = false, columnDefinition = "VARCHAR")
    @field:Enumerated(EnumType.STRING)
    var status: UserOrderMenuStatus? = UserOrderMenuStatus.REGISTERED,

    @field:CreationTimestamp
    @field:Column(updatable = false)
    private var createdAt: LocalDateTime? = null,

    @field:UpdateTimestamp
    private var updatedAt: LocalDateTime? = null,
) {
}